dotnet/src/Connectors/Connectors.MistralAI/MistralAIPromptExecutionSettings.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.Connectors.MistralAI;
+
+/// <summary>
+/// Mistral Execution Settings.
+/// </summary>
+[JsonNumberHandling(JsonNumberHandling.AllowReadingFromString)]
+public sealed class MistralAIPromptExecutionSettings : PromptExecutionSettings
+{
+    /// <summary>
+    /// Default: 0.7
+    /// What sampling temperature to use, between 0.0 and 1.0. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic.
+    /// </summary>
+    /// <remarks>
+    /// We generally recommend altering this or top_p but not both.
+    /// </remarks>
+    [JsonPropertyName("temperature")]
+    public double Temperature
+    {
+        get => this._temperature;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._temperature = value;
+        }
+    }
+
+    /// <summary>
+    /// Default: 1
+    /// Nucleus sampling, where the model considers the results of the tokens with top_p probability mass.So 0.1 means only the tokens comprising the top 10% probability mass are considered.
+    /// </summary>
+    /// <remarks>
+    /// We generally recommend altering this or temperature but not both.
+    /// </remarks>
+    [JsonPropertyName("top_p")]
+    public double TopP
+    {
+        get => this._topP;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._topP = value;
+        }
+    }
+
+    /// <summary>
+    /// Default: null
+    /// The maximum number of tokens to generate in the completion.
+    /// </summary>
+    /// <remarks>
+    /// The token count of your prompt plus max_tokens cannot exceed the model's context length.
+    /// </remarks>
+    [JsonPropertyName("max_tokens")]
+    public int? MaxTokens
+    {
+        get => this._maxTokens;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._maxTokens = value;
+        }
+    }
+
+    /// <summary>
+    /// Default: false
+    /// Whether to inject a safety prompt before all conversations.
+    /// </summary>
+    [JsonPropertyName("safe_prompt")]
+    public bool SafePrompt
+    {
+        get => this._safePrompt;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._safePrompt = value;
+        }
+    }
+
+    /// <summary>
+    /// Default: null
+    /// The seed to use for random sampling. If set, different calls will generate deterministic results.
+    /// </summary>
+    [JsonPropertyName("random_seed")]
+    public int? RandomSeed
```

Comment:
How are we thinking about naming from the perspective of unifying same-purpose properties across disparate settings types?

Reply:
We don't do anything explicitly to unify property names. @matthewbolanos what are your thoughts on this?

Reply:
We don't have a definitive plan here. We discussed with Matthew and will wait on some use cases before deciding how to proceed.